### PR TITLE
Correct cloud_cover_to_transmittance_linear documentation

### PR DESCRIPTION
**Pull request by JoLo90**
Corrects the documentation of the **cloud_cover_to_transmittance_linear** function. The current return is documented as the ghi instead of the transmittance. The information "percentage" has been added to the cloud cover and "ratio" to the transmittance.

 - [x] Closes [pvlib#1367](https://github.com/pvlib/pvlib-python/issues/1367)
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

File: docs/sphinx/source/whatsnew/v0.9.1.rst
Comment (kanderso-nrel) [inline docs/sphinx/source/whatsnew/v0.9.1.rst:33]:
```suggestion
  (:issue:`1367`, :pull:`1370`)
```

**Review by kandersolar — COMMENTED** [review]
Thanks @JoLo90!  The coverage and pytest failures can be ignored here.  One suggestion below, otherwise LGTM

File: docs/sphinx/source/whatsnew/v0.9.1.rst
Comment (JoLo90) [reply docs/sphinx/source/whatsnew/v0.9.1.rst]: thanks @kanderso-nrel it has been added

**Review by AdamRJensen — COMMENTED** [review]
Looks good!

File: pvlib/forecast.py
Comment (cwhanse) [inline pvlib/forecast.py:527]:
```suggestion
            Determines the maximum transmittance. [unitless]
```

**Review by cwhanse — COMMENTED** [review]
Thanks for this contribution, @JoLo90. Some suggested edits.

**Review by cwhanse — APPROVED** [review]

**Comment by cwhanse** [comment]
@pvlib/pvlib-maintainer I will merge tomorrow unless additional comments are made.

**Pull request merged**
